pettingzoo/utils/wrappers/base_parallel.py

```diff
@@ -1,40 +1,26 @@
 from __future__ import annotations
 
-import warnings
-
 import gymnasium.spaces
 import numpy as np
-from gymnasium.utils import seeding
 
 from pettingzoo.utils.env import ActionType, AgentID, ObsType, ParallelEnv
 
 
 class BaseParallelWrapper(ParallelEnv[AgentID, ObsType, ActionType]):
     def __init__(self, env: ParallelEnv[AgentID, ObsType, ActionType]):
+        super().__init__()
         self.env = env
 
-        self.metadata = env.metadata
-        try:
-            self.possible_agents = env.possible_agents
-        except AttributeError:
-            pass
-
-        # Not every environment has the .state_space attribute implemented
-        try:
-            self.state_space = (
-                self.env.state_space  # pyright: ignore[reportGeneralTypeIssues]
-            )
-        except AttributeError:
-            pass
+    def __getattr__(self, name: str):
+        """Returns an attribute with ``name``, unless ``name`` starts with an underscore."""
+        if name.startswith("_"):
+            raise AttributeError(f"accessing private attribute '{name}' is prohibited")
+        return getattr(self.env, name)
 
     def reset(
         self, seed: int | None = None, options: dict | None = None
     ) -> tuple[dict[AgentID, ObsType], dict[AgentID, dict]]:
-        self.np_random, _ = seeding.np_random(seed)
-
-        res, info = self.env.reset(seed=seed, options=options)
-        self.agents = self.env.agents
-        return res, info
+        return self.env.reset(seed=seed, options=options)
```

Comment:
Is the reason for not setting self np random that you want that to be handled on the environment itself and not messed with by the wrapper

Reply:
Yup, makes no sense to me that it is the wrappers that set the random. I wanted to move it to `BaseAEC.reset()` at first but then figured that it would be overriden all the time...

Reply:
Yeah makes sense